hoot-core-test/src/test/cpp/hoot/core/ops/MapCropperTest.cpp

```diff
@@ -243,6 +249,36 @@ class MapCropperTest : public CppUnit::TestFixture
     HOOT_STR_EQUALS("Env[0:-1,0:-1]", cropper._envelope.toString());
   }
 
+  void runMultiPolygonTest()
+  {
+    shared_ptr<OsmMap> map(new OsmMap());
+    OsmMapReaderFactory::read(map, "test-files/MultipolygonTest.osm",true);
+
+    Envelope env(0.30127,0.345,0.213,0.28154);
+
+    MapCropper::crop(map, env);
+
+    const WayMap ways = map->getWays();
+    HOOT_STR_EQUALS(2, ways.size());
+
+    int count = 0;
+    for (WayMap::const_iterator it = ways.begin(); it != ways.end(); it++)
+    {
+      const shared_ptr<Way>& w = it->second;
+      shared_ptr<LineString> ls = ElementConverter(map).convertToLineString(w);
+      const Envelope& e = *(ls->getEnvelopeInternal());
+      if (count == 0)
+      {
+        HOOT_STR_EQUALS("Env[0.303878:0.336159,0.220255:0.270199]", e.toString());
+      }
+      else
+      {
+        HOOT_STR_EQUALS("Env[0.314996:0.328946,0.231514:0.263127]", e.toString());
+      }
+      count++;
+    }
+  }
```

Comment:
@mingjiesu This is only testing to see if the ways are cropped properly. Can you add a check to see if the relations/polygons are being cropped properly too? E.g. Crop and then check the area of each polygon?